### PR TITLE
[ADF-4278][ADF-4277] - Should fetch the preselect users

### PR DESCRIPTION
**Pull request by gmandakini**
**Please check if the PR fulfills these requirements**

> - [ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
https://issues.alfresco.com/jira/browse/ADF-4278
https://issues.alfresco.com/jira/browse/ADF-4277

### CODE REVIEW

**Comment by gmandakini** [comment]
The tests are failing as the functionality has broken as part of this PR being merged : https://github.com/Alfresco/alfresco-ng2-components/pull/4526/files
It will be fixed and working again as part of this PR : https://github.com/Alfresco/alfresco-ng2-components/pull/4474

File: lib/testing/src/lib/core/models/user.model.ts [hunk 1]
```diff
@@ -20,9 +20,10 @@ import { StringUtil } from '../string.util';
 export class UserModel {
 
     firstName: string = StringUtil.generateRandomString();
-    lastName: string = StringUtil.generateRandomString();
+    lastName: string = StringUtil.generateRandomString() + 'LastName';
```
Comment (eromano) [inline lib/testing/src/lib/core/models/user.model.ts]: why do we need always this LastName?

Comment (gmandakini) [reply lib/testing/src/lib/core/models/user.model.ts]: so that we can search on a common lastname and assert that we are only seeing users that we expect to see in the dropdown. Basically to have a common suffix that we can search on.

File: e2e/process-services-cloud/people-group-cloud-component.e2e.ts [hunk 2]
```diff
@@ -170,6 +181,127 @@ describe('People Groups Cloud Component', () => {
             groupCloudComponentPage.checkSelectedGroup(`${groupNoRole.name}`);
         });
 
+        it('[C305033] Should fetch the preselect users based on the Validate flag set to True or False', () => {
```
Comment (cristinaj) [inline e2e/process-services-cloud/people-group-cloud-component.e2e.ts]: Can be break this test in smaller test case? It's hard to follow the test. I would break it by component's properties: mode, validate, etc.

File: e2e/process-services-cloud/people-group-cloud-component.e2e.ts [hunk 3]
```diff
@@ -170,6 +181,127 @@ describe('People Groups Cloud Component', () => {
...
+            peopleGroupCloudComponentPage.clickPeopleCloudSingleSelection();
+            peopleGroupCloudComponentPage.clickPreselectValidation();
+            peopleGroupCloudComponentPage.checkPreselectValidationIsChecked();
+            peopleGroupCloudComponentPage.enterPeoplePreselect(`[{"id":"${noRoleUser.id}"}]`);
+            browser.sleep(100);
```
[the remark this answers is not in the document]
Comment (gmandakini) [reply e2e/process-services-cloud/people-group-cloud-component.e2e.ts]: I couldn't find a way to get it working without the sleep. Please advise.

File: e2e/pages/adf/demo-shell/process-services/peopleGroupCloudComponentPage.ts [hunk 4]
```diff
@@ -27,11 +27,17 @@ export class PeopleGroupCloudComponentPage {
...
     peopleRoleInput = element(by.css('input[data-automation-id="adf-people-roles-input"]'));
+    peopleAppInput = element(by.css('input[data-automation-id="adf-people-app-input"]'));
     peoplePreselect = element(by.css('input[data-automation-id="adf-people-preselect-input"]'));
     groupRoleInput = element(by.css('input[data-automation-id="adf-group-roles-input"]'));
+    groupAppInput = element(by.css('input[data-automation-id="adf-group-app-input"]'));
     groupPreselect = element(by.css('input[data-automation-id="adf-group-preselect-input"]'));
     peopleCloudComponentTitle = element(by.cssContainingText('mat-card-title', 'People Cloud Component'));
     groupCloudComponentTitle = element(by.cssContainingText('mat-card-title', 'Groups Cloud Component'));
+    preselectValidation = element(by.css('mat-checkbox.adf-preselect-value'));
+    assigneeField = element(by.css('input[data-automation-id="adf-people-cloud-search-input"]'));
```
[the remark this answers is not in the document]
Comment (gmandakini) [reply e2e/pages/adf/demo-shell/process-services/peopleGroupCloudComponentPage.ts]: I am moving assignee filed, preselectValidation is still part of peopleGroupCloudComponentPage.
